### ISSUE TITLE
x/build: freebsd-amd64-race logs for x/tools marked as failing, but do not include error messages

### ISSUE THREAD

**Issue by bcmills** [golang/go#49347]
(Note that the below regexp is not specific to freebsd or to x/tools. It is just looking for logs that contain only non-error lines.)

`greplogs --dashboard -md -l -e '(?m)\A[^ ]+ at.*\n\n(?:(?:[:][:] Running .+\n\n)|(?:go: downloading [^ ]+ v[^ ]+\n)|(?:ok\s+[^ ]+\s+[^ ]+s\n)|(?:[?]\s+[^ ]+\s+\[no test files\]\n))+\z'`

[2021-11-04T07:05:31-84e69e7-2622235/freebsd-amd64-race](https://build.golang.org/log/7938f65467eb2d8b551e4e4f2814071663fe65e8)
[2021-11-03T23:49:25-84e69e7-fa19146/freebsd-amd64-race](https://build.golang.org/log/dac94bc58b344c4a40d2616547a86f91b224f9aa)
[2021-11-03T19:32:33-3b1a674-74f99d0/freebsd-amd64-race](https://build.golang.org/log/ccbc551ba48d50c56eb3a25bd7967b76250e15cd)
[2021-11-03T00:49:50-714668c-a4b2c57/freebsd-amd64-race](https://build.golang.org/log/30635abe230b3f4ee3a335edc58b26f30a637469)
[2021-11-02T23:44:58-714668c-a0f373c/freebsd-amd64-race](https://build.golang.org/log/9a8c09cbf6ebcc477ff32c8a878ba957046d5168)
[2021-11-02T16:40:04-058ed05-8c8baad/freebsd-amd64-race](https://build.golang.org/log/5df09552cdf2e8035aa1fdf4477b8a1a613d43e1)
[2021-11-02T00:12:17-ee08195-81fea0b/freebsd-amd64-race](https://build.golang.org/log/53850b6012cd121aeab4776bda684267af35342d)
[2021-10-28T16:54:58-c8ad2e1-6bd0e7f/freebsd-amd64-race](https://build.golang.org/log/b6e969c9e7ab8ea12285d9b2eef0f4266e479723)
[2021-10-26T01:18:46-f916b54-49fda9e/freebsd-amd64-race](https://build.golang.org/log/d842e4b8538c93865be715d7c3b0c0c044eb763a)
[2021-10-25T14:51:39-fc8b4ca-1ed060c/freebsd-amd64-race](https://build.golang.org/log/ab741fa635ed0585a2f578b40808f5f9e9c1774a)
[2021-10-17T02:23:16-98f6e03-fa7d11a/freebsd-amd64-race](https://build.golang.org/log/37830bbcd5b802d02d03093fea4f4615964f6e32)
[2021-10-16T16:27:40-98f6e03-640a49b/freebsd-amd64-race](https://build.golang.org/log/78a0f96a28f9530866cdcddfbaf02fde981eed8f)
[2021-10-16T14:50:21-98f6e03-680caf1/freebsd-amd64-race](https://build.golang.org/log/373949b68abd31644d0647126941af1a21e6e35b)
[2021-10-15T20:34:15-98f6e03-1b072b3/freebsd-amd64-race](https://build.golang.org/log/96bc62fb59d5ad37dfd7e951eb04106c1c50a303)
[2021-10-15T14:09:01-98f6e03-1cbec68/freebsd-amd64-race](https://build.golang.org/log/ad01a44b7d6a2f6de0c3fbb72ffa17cb66950114)
[2021-10-15T02:45:22-288d0b3-2ac3bdf/freebsd-amd64-race](https://build.golang.org/log/aae4867a4c1e61b37fb74b53ab4ec55d0713c9d4)
[2021-10-15T02:02:50-ce04ca3-3da0ff8/freebsd-amd64-race](https://build.golang.org/log/de37aa32306bacd0c0681e712b8fc278d7713e2d)
[2021-10-15T01:35:56-ce04ca3-0c45ed0/freebsd-amd64-race](https://build.golang.org/log/fffc0454e23c6fee6f8cb7501d9a7f2f46d83d8a)
[2021-10-14T23:32:09-3a269dc-9e8ed86/freebsd-amd64-race](https://build.golang.org/log/fce238861826a3edf1a92cce10b0a55df1b0c0ea)
[2021-10-14T18:44:37-e69ba9d-a37bebc/freebsd-amd64-race](https://build.golang.org/log/9b788704ef1bef91fdd17c4ebcb29687cddd2b93)

CC @golang/release

**Comment by bcmills** [comment]
I think the process may be being killed due to memory exhaustion.

The `freebsd-amd64-race` builder appears to be running on an `e2-highcpu-16` builder, while the normal `linux-amd64-race` builder is running on an `e2-standard-4`. (https://play.golang.org/p/5LXUgwHEZPn)

According to https://cloud.google.com/compute/vm-instance-pricing, `e2-standard-4` has 4 CPUs and 16GB memory, while `e2-highcpu-16` has 16 CPUs and 16GB.

Since the default GOMAXPROCS is based on the CPU count, and the default test parallelism is based on GOMAXPROCS, I suspect that the increased CPU-to-RAM ratio on the `highcpu` host is causing it to run too many `gopls` tests in parallel and run out of RAM.

@findleyr, @golang/release: should we address this by adjusting the host shape, or by adding some kind of additional limiting to the test, or explicitly setting GOMAXPROCS on this builder (which would possibly reduce its testing throughput), or something else entirely?

**Comment by findleyr** [comment]
Comparing these two recent builds:
Fail: https://build.golang.org/log/4d807f4a6b0659eea741ece539bd6a39a4fc0c1a
Success: https://build.golang.org/log/2d9892c7aaf14942405fb6b82ae24234b1807bec

We can see that the failure is not finishing all x/tools tests --  the last test reported is go/loader.
The next test in the success is go/packages, so I suspect this is failing in go/packages (@bcmills is that a viable conclusion?)

**Comment by findleyr** [comment]
If this is OOMing in go/packages (which we suspect may lead to this type of silent failure), then I can think of two possible sources:
 - Increase memory footprint or allocation rate in go/types.
 - Parallelization of those tests in https://golang.org/cl/332351

Looking specifically at some of the go/packages tests, I am also curious about why the tests are so slow and memory intensive, based on what they are doing. e.g. TestLoadDifferentPatterns takes 3s, but that seems extremely slow based on my understanding of our type-checking speed and the scope of that test.

**Comment by bcmills** [comment]
> The next test in the success is go/packages, so I suspect this is failing in go/packages (@bcmills is that a viable conclusion?)

I think so — well spotted!


> I can think of two possible sources:
> * Increase memory footprint or allocation rate in go/types.
> * Parallelization of those tests in https://golang.org/cl/332351

That is an interesting theory! But that change was merged in July, and the first of these tests started in October.

My leading theory is still machine shape (this builder has a very high CPU-to-RAM ratio), but either or both of those could be aggravating factors as well.

**Comment by bcmills** [comment]
This builder is still failing multiple times per day on `x/tools` in multiple different packages. We could theoretically fix the tests for each of those packages to scale by RAM, but at this point I'd really just rather make a pointwise change and be done with it.

@golang/release: could we try scaling this builder back to a more balanced configuration?

`greplogs --dashboard -md -l -e \\Afreebsd-amd64-race -E . --since=2022-01-28`

[2022-01-31T20:30:41-492cf3a-41f485b/freebsd-amd64-race](https://build.golang.org/log/d7758ad1f7746a5171f7bad8230b3e5d2f254103)
[2022-01-31T13:43:52-939c2c0-41f485b/freebsd-amd64-race](https://build.golang.org/log/48a6c8df172384109dc25011d6f19cb767b3e259)
[2022-01-29T19:58:17-939c2c0-1226547/freebsd-amd64-race](https://build.golang.org/log/54e445a5a282cc2236c08d3c46076096ce43890f)
[2022-01-29T03:53:30-939c2c0-5fd8c9b/freebsd-amd64-race](https://build.golang.org/log/498b0bf20299d0f5ea6899fa2d3295108ee5666f)
[2022-01-28T22:17:23-939c2c0-654d5f4/freebsd-amd64-race](https://build.golang.org/log/410e2704a1cce8df14cb408c990a49b417fe8745)
[2022-01-28T20:07:54-939c2c0-8314544/freebsd-amd64-race](https://build.golang.org/log/24a2ef97f340c1b07c16fd38e437b3c215cd9847)

**Comment by bcmills** [comment]
Lots of recent failures on this builder, but — crucially! — none missing error messages. I'm calling this fixed. (We can always reopen — or open a new issue — if that turns out to be wrong.)

`greplogs -l -e \\Afreebsd-amd64-race -E . --since=2022-05-01`
[2022-05-05T18:49:32-983906f/freebsd-amd64-race](https://build.golang.org/log/5ad3f62cdd3bb3c4eed2f6894f15b1617df470a4)
[2022-05-05T18:49:25-a7ab208/freebsd-amd64-race](https://build.golang.org/log/c0da2d314bfb6739dd91210776cea610f6bc0042)
[2022-05-05T18:49:17-5073c1c/freebsd-amd64-race](https://build.golang.org/log/9a9ce983e4f82025afc6f02bd91c2e70f98c0f88)
[2022-05-04T21:35:22-a5481fb-aeb933d/freebsd-amd64-race](https://build.golang.org/log/ab2caa3ef5d8af43c6943cd643717f9ec6374713)
[2022-05-04T21:03:31-a5481fb-2f23364/freebsd-amd64-race](https://build.golang.org/log/11a31b616b48c634aeff480e0639c7186704ac78)
[2022-05-04T20:30:17-a5481fb-e1e056f/freebsd-amd64-race](https://build.golang.org/log/40c5632ac2ba6389c55018e121b74c381c7cec01)
[2022-05-04T20:20:28-a5481fb-fbb47e8/freebsd-amd64-race](https://build.golang.org/log/bfb3b02b9a15e3b857086834407895900ac7c01c)
[2022-05-04T20:06:32-a5481fb-fd6ef06/freebsd-amd64-race](https://build.golang.org/log/cbc4fb5042c662943ee4297e203b4c3ec3fb5eff)
[2022-05-04T18:54:31-a5481fb-037b209/freebsd-amd64-race](https://build.golang.org/log/6d119b885297e08858ff35839246c77a7292a5db)
[2022-05-04T18:51:19-1b0f9fb/freebsd-amd64-race](https://build.golang.org/log/000c295245a59a50a76b9fe68758cf8fe4815fa2)
[2022-05-04T18:05:18-a5481fb-ffe48e0/freebsd-amd64-race](https://build.golang.org/log/6e771f7a4dd3cddd37df821a5ece50e499108b9b)
[2022-05-04T18:05:18-78819d0-ffe48e0/freebsd-amd64-race](https://build.golang.org/log/4ef83371353d29dc7868d7f6cf39f2953eb82d3e)
[2022-05-04T16:13:20-f52b4ec/freebsd-amd64-race](https://build.golang.org/log/4473336e12946223114e3fb3baf265b80edf23b5)
[2022-05-03T20:54:41-d24c65d/freebsd-amd64-race](https://build.golang.org/log/668b652d86d49a31f31c655f0e45c295a841747d)
[2022-05-03T20:49:59-89c0dd8/freebsd-amd64-race](https://build.golang.org/log/03481cb57dbf48e06caac6747895d698c0639078)
[2022-05-03T20:49:46-2c0a988/freebsd-amd64-race](https://build.golang.org/log/03923d6cab9a590bade107b70d4d5695dd0abcd0)
[2022-05-03T15:14:09-f01c20b/freebsd-amd64-race](https://build.golang.org/log/a8e5974effc1e14a810a5b643365dedf61bd39ef)
[2022-05-03T15:13:53-91f8630/freebsd-amd64-race](https://build.golang.org/log/56d8411d209454e4600c91bf3175bd39eb06153d)
[2022-05-03T15:13:45-b4d8114/freebsd-amd64-race](https://build.golang.org/log/86fa392181fc840af678661df3140204c6f6a39e)
[2022-05-03T15:13:35-7e4bc74/freebsd-amd64-race](https://build.golang.org/log/b5f5a6f79e4f705139179bb29046d3e40b38d55a)
[2022-05-03T15:13:21-973dcbb/freebsd-amd64-race](https://build.golang.org/log/c3bad1a54cb330e74abbdd767e9d6762ca09c62e)
[2022-05-03T15:12:52-129dcb7/freebsd-amd64-race](https://build.golang.org/log/1a254dd163cd6af74d72a6497a6bb20cce8c5a12)
[2022-05-03T15:12:45-473c996/freebsd-amd64-race](https://build.golang.org/log/c4d72d3a651c60547c271ae156dddb88f9aba955)
[2022-05-03T15:12:38-375d696/freebsd-amd64-race](https://build.golang.org/log/45236034adc0d0c1119bd4237a08dcd956407634)
[2022-05-03T15:12:31-d36d5bd/freebsd-amd64-race](https://build.golang.org/log/a3bf0847e3166198a3a9b01ae4f7eb9f92b02e24)
[2022-05-03T15:12:21-4649a43/freebsd-amd64-race](https://build.golang.org/log/34b6714b2af2aafb377d2c60b41a870b059a2155)
[2022-05-03T15:12:12-85d4664/freebsd-amd64-race](https://build.golang.org/log/79122726f04927252bd44789001bbdc493026cd5)
[2022-05-03T15:12:04-986a310/freebsd-amd64-race](https://build.golang.org/log/b37d3f8caeb3cd535d1cd56c47cc7405817b9f21)
[2022-05-03T15:11:55-0feebe6/freebsd-amd64-race](https://build.golang.org/log/6a8988a1151cad68124cc107e53a7313fc8bf7c1)
[2022-05-03T15:11:42-01359b4/freebsd-amd64-race](https://build.golang.org/log/7472f97c08da0085fd5b8d94e47ad41228d722b1)
[2022-05-02T21:05:53-04fc2ba-a887579/freebsd-amd64-race](https://build.golang.org/log/377bc493565b75a8b3825b69e0428c053f4abf93)
